Modules/Filtering/ImageGrid/test/itkResampleImageTest2s.cxx

```diff
@@ -0,0 +1,267 @@
+    }
+
+  // Assign an affine transform that returns
+  // false for IsLinear() instead of true, to force
+  // the filter to use the NonlinearThreadedGenerateData method
+  // instead of LinearThreadedGenerateData. This will test that
+  // we get the same results for both methods.
+  std::cout << "Test with NonlinearAffineTransform." << std::endl;
+  NonlinearAffineTransformType::Pointer nonlinearAffineTransform =
+                                    NonlinearAffineTransformType::New();
+
+  nonlinearAffineTransform->Scale(2.0);
+  resample->SetTransform( nonlinearAffineTransform );
+  writer2->SetInput( resample->GetOutput() );
+  try
+    {
+    writer2->SetNumberOfStreamDivisions(8); //demand splitting into 8 pieces for streaming, but faked non-linearity will disable streaming
+    writer2->Update();
+    }
+  catch( itk::ExceptionObject & excp )
+    {
+    std::cerr << excp << std::endl;
+    return EXIT_FAILURE;
+    }
+
+  // Instead of using the default pixel when sampling outside the input image,
+  // we use a nearest neighbor extrapolator.
+  resample->SetTransform( affineTransform );
+  resample->SetExtrapolator( extrapolator );
+  writer3->SetInput( resample->GetOutput() );
+  std::cout << "Test with nearest neighbor extrapolator, affine transform." << std::endl;
+  try
+    {
+    writer3->SetNumberOfStreamDivisions(8); //split into 8 pieces for streaming.
+    writer3->Update();
+    }
+  catch( itk::ExceptionObject & excp )
+    {
+    std::cerr << excp << std::endl;
+    return EXIT_FAILURE;
+    }
+
+  // Instead of using the default pixel when sampling outside the input image,
+  // we use a nearest neighbor extrapolator.
+  resample->SetTransform( nonlinearAffineTransform );
+  writer4->SetInput( resample->GetOutput() );
+  std::cout << "Test with nearest neighbor extrapolator, nonlinear transform." << std::endl;
+  try
+    {
+    writer4->SetNumberOfStreamDivisions(8); //demand splitting into 8 pieces for streaming, but faked non-linearity will disable streaming
+    writer4->Update();
+    }
+  catch( itk::ExceptionObject & excp )
+    {
+    std::cerr << excp << std::endl;
+    return EXIT_FAILURE;
+    }
+
+  // Check UseReferenceImage methods
+  resample->UseReferenceImageOff();
+  if( resample->GetUseReferenceImage() )
+    {
+    std::cerr << "GetUseReferenceImage() or UseReferenceImageOff() failed ! ";
+    std::cerr << std::endl;
+    return EXIT_FAILURE;
+    }
+
+  // Check UseReferenceImage methods
+  resample->UseReferenceImageOn();
+  if( !resample->GetUseReferenceImage() )
+    {
+    std::cerr << "GetUseReferenceImage() or UseReferenceImageOn() failed ! ";
+    std::cerr << std::endl;
+    return EXIT_FAILURE;
+    }
+
+  // Check UseReferenceImage methods
+  resample->SetUseReferenceImage( false );
+  if( resample->GetUseReferenceImage() )
+    {
+    std::cerr << "GetUseReferenceImage() or SetUseReferenceImage() failed ! ";
+    std::cerr << std::endl;
+    return EXIT_FAILURE;
+    }
```

Comment:
All the above calls can be substituted by calling the `TEST_SET_GET_BOOLEAN()` macro. Best would be to add it as a parameter of the test and add two tests, one exercising the `On` case, the other exercising the `Off` case.

Reply:
removed in https://github.com/romangrothausmann/ITK/commit/0c738df46521afcaa0366a57e9f6a712db18a851

Reply:
Why were they removed/why was `TEST_SET_GET_BOOLEAN()` not used?

The boolean functions are still present, so it is indeed a necessary check for the good health of the class to test them. And as said, even better adding the argument as a parameter and exercising both `true` and `false` conditions.

Reply:
Thanks for looking at the new commit (you are a really very careful reader!). I removed them because they are an unintended remainder (have nothing to do with streaming) from `itkResampleImageTest2.cxx` (which I used as a base) and are still tested there: https://github.com/romangrothausmann/ITK/blob/0c738df46521afcaa0366a57e9f6a712db18a851/Modules/Filtering/ImageGrid/test/itkResampleImageTest2.cxx#L217-L242

Reply:
I see. Thanks.